synapse/federation/transport/server.py

```diff
@@ -338,6 +338,12 @@ async def new_func(request, *args, **kwargs):
                 if origin:
                     with ratelimiter.ratelimit(origin) as d:
                         await d
+                        if request._disconnected:
+                            logger.warning(
+                                "client disconnected before we started processing "
+                                "request"
+                            )
+                            return -1, None
```

Comment:
-1 just because this is a garbage response and it doesn't really matter?

This seems similar to #7927 where `200, {}` was returned. Why the difference?

Reply:
errr good question.

`None` ensures that we don't call `respond_with_json`.

Reply:
Should we be using it in the other place too then? 😄 Either way I think this is OK!

Reply:
yes, probably. I think this this the better way. we could clean up #7927 for consistency but I don't think it's a big deal really.